docs/endpoints/config_reference.rst

```diff
@@ -0,0 +1,297 @@
+   25729 ...     \_ Globus Compute Endpoint (..., four_workers_100_tasks)
+   26339 ... \_ Globus Compute Endpoint (..., three_threads)
+
+   $ ps wm --forest | grep -A2 three_threads
+   26339 ... Globus Compute Endpoint (..., three_threads)
+       - ... -
+       - ... -
+
+Per usual Python semantics, the ``ThreadPoolEngine`` (``ThreadPoolExecutor`` under the
+hood) is typically best for I/O oriented workflows, while the ``ProcessPoolEngine``
+(``ProcessPoolExecutor`` under the hood) will be a better fit for CPU-intensive tasks.
+But both engines will only run tasks *on the endpoint host machine*.  If the endpoint is
+strictly limited to a single host (e.g., a home desktop, an idle workstation), then
+these engines may be the simplest option.
+
+For running in a multi-node setup (e.g., clusters, with scheduling software like PBS or
+Slurm), the ``GlobusComputeEngine`` enables much more concurrency.  This engine has more
+options and is similarly more complicated to configure.  A rough equivalent to the
+``ProcessPoolEngine`` example would be:
+
+.. code-block:: yaml
+   :caption: ``~/.globus_compute/my_first_cluster_setup/config.yaml``
+
+   engine:
+     type: GlobusComputeEngine
+     provider:
+       type: LocalProvider
+       max_blocks: 4
+
+Whereas the ``ThreadPoolEngine`` and ``ProcessPoolEngine`` wrappers have an implicit
+approach to managing the compute resources (the `process model`_), the
+``GlobusComputeEngine`` requires explicit knowledge of the local topology.  The
+``provider`` stanza chooses the Parsl mechanism by which to communicate with the local
+site.  If the workers will be on the same host as the endpoint, |LocalProvider|_ is
+appropriate.  But if workers will be on cluster nodes, those resources will be accessed
+via the site-specific scheduler, and communication between the endpoint and the workers
+will occur via a site-specific network interface.
+
+`Parsl implements a number of providers`_, so we will not describe them here.  Instead,
+we present a single example, and then suggest reading through the rest of the
+:doc:`endpoint configuration examples <endpoint_examples>` and Parsl's documentation
+while keeping your project needs in mind.
+
+The University of Chicago's Midway cluster uses `Slurm`_ as the batch scheduler, so the
+following example configuration chooses Parsl's |SlurmProvider|_.  The Slurm batch
+scheduler requires an allocation to debit for each job submission, specified for
+|SlurmProvider|_ via ``account``.  On each acquired cluster node, the ``worker_init``
+is a set of shell-script lines that will be run prior to starting the worker; this
+example loads the site-specific module named ``Anaconda``, and then activates the
+``compute-env`` environment.  When sending jobs to the batch scheduler, this requests
+only a single node at a time (``nodes_per_block``), from the partition of nodes named
+``caslake`` (``partition``), does not have more than 1 active job (``max_blocks``), and
```

Comment:
"active" I suspect is ambiguous. I'd add some clarification here, say: `active (pending + running) job`.